torch/distributed/_tensor/_utils.py

```diff
@@ -224,3 +224,68 @@ def compute_local_stride(
     return tuple(
         global_stride[i] // stride_divisors[i] for i in range(len(global_stride))
     )
+
+
+def compute_padded_and_unpadded_local_shape(
+    global_shape: ShapeType, mesh: DeviceMesh, placements: Sequence[Placement]
```

Comment:
We can later change to DTensorSpec when we are ready to convert to static padding. Currently, I am not testing the util through a DTensor, since we haven't introduced static padding in DTensor yet.